src/test/java/org/opensearch/security/http/OnBehalfOfAuthenticatorTest.java

```diff
@@ -0,0 +1,400 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.http;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.crypto.SecretKey;
+
+import com.google.common.io.BaseEncoding;
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.user.AuthCredentials;
+import org.opensearch.security.util.FakeRestRequest;
+
+public class OnBehalfOfAuthenticatorTest {
+    final static String clusterName = "cluster_0";
+    final static String enableOBO = "true";
+    final static String disableOBO = "false";
+    final static String claimsEncryptionKey = RandomStringUtils.randomAlphanumeric(16);
+
+    final static String signingKey =
+        "This is my super safe signing key that no one will ever be able to guess. It's would take billions of years and the world's most powerful quantum computer to crack";
+    final static String signingKeyB64Encoded = BaseEncoding.base64().encode(signingKey.getBytes(StandardCharsets.UTF_8));
+    final static SecretKey secretKey = Keys.hmacShaKeyFor(signingKeyB64Encoded.getBytes(StandardCharsets.UTF_8));
+
+    @Test
+    public void testNoKey() throws Exception {
+
+        try {
+            final AuthCredentials credentials = extractCredentialsFromJwtHeader(
+                null,
+                claimsEncryptionKey,
+                Jwts.builder().setIssuer(clusterName).setSubject("Leonard McCoy"),
+                false
+            );
+            Assert.fail("Expected a RuntimeException");
+        } catch (RuntimeException e) {
+            Assert.assertTrue(e.getMessage().contains("Unable to find on behalf of authenticator signing key"));
+        }
+    }
+
+    @Test
+    public void testEmptyKey() throws Exception {
+
+        try {
+            final AuthCredentials credentials = extractCredentialsFromJwtHeader(
+                null,
+                claimsEncryptionKey,
+                Jwts.builder().setIssuer(clusterName).setSubject("Leonard McCoy"),
+                false
+            );
```

Comment:
Use `assertThrows` to expect an exception, you can refine the expected type to be very granular too

```suggestion
            final Exception e = assertThrows(Exception.class, () -> extractCredentialsFromJwtHeader(
                null,
                claimsEncryptionKey,
                Jwts.builder().setIssuer(clusterName).setSubject("Leonard McCoy"),
                false
            ));
            assertThat(e.getMessage(),...);
```

Reply:
Good call out. I have changed them with `Assert.assertThrows`